src/index.ts

```diff
@@ -13,9 +13,11 @@ import { resolveRootPath } from './options/rootPath';
 import { purge } from './purge';
 import { Directory } from './interfaces/directory.interface';
 
+export type { Arguments };
+
 // TODO: Document how users can call this from their own code without using the CLI.
 // TODO: We might need to do some parameter validation for that.
-export function Barrelsby(args: Arguments) {
+export async function Barrelsby(args: Arguments): Promise<void> {
```

Comment:
Should this be a separate function that calls the main `Barrelsby` function? @bencoveney thoughts?